SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultWorkerJsonGenerator.cpp

```diff
@@ -46,19 +65,43 @@ bool GenerateAllDefaultWorkerJsons(bool& bOutRedeployRequired)
 
 	if (const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>())
 	{
-		const FName WorkerTypes[] = { SpatialConstants::DefaultServerWorkerType, SpatialConstants::RoutingWorkerType,
-									  SpatialConstants::StrategyWorkerType };
-		for (auto Worker : WorkerTypes)
+		// Create an array of worker types with a bool signifying whether the associated worker json should exist or not.
+		// Then correct the file system state so it matches our expectations.
+		TArray<TPair<FName, bool>> WorkerTypes;
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::DefaultServerWorkerType, true));
+		const bool bRoutingWorkerEnabled = SpatialGDKSettings->CrossServerRPCImplementation == ECrossServerRPCImplementation::RoutingWorker;
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::RoutingWorkerType, bRoutingWorkerEnabled));
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::StrategyWorkerType, SpatialGDKSettings->bRunStrategyWorker));
+
+		for (const auto& Pair : WorkerTypes)
 		{
-			FString JsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.json"), *Worker.ToString()));
-			if (!FPaths::FileExists(JsonPath))
+			bool bShouldFileExist = Pair.Value;
+			FString WorkerType = Pair.Key.ToString();
+			FString JsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.json"), *WorkerType));
+			FString UnusedJsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.UNUSED.json"), *WorkerType));
+			const bool bFileExists = FPaths::FileExists(JsonPath);
+			const bool bUnusedFileExists = FPaths::FileExists(UnusedJsonPath);
+
+			if (!bFileExists && bShouldFileExist)
 			{
 				UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Verbose, TEXT("Could not find worker json at %s"), *JsonPath);
```

Comment:
We should update this to say it will generate it now.

Reply:
It does say so within the generate function